pkg/cdi/cache.go

```diff
@@ -149,6 +206,13 @@ func (c *Cache) Refresh() error {
 	return nil
 }
 
+// RefreshIfRequired triggers a refresh with auto-refresh if Spec dirs have appeared.
+func (c *Cache) refreshIfRequired() {
+	if c.autoRefresh && c.watch.update(c.dirErrors) {
```

Comment:
See my earlier comment about possibly adding a `force` argument here.

Reply:
Rewrote, distilling the former manual `Refresh()` logic here, too.